pkg/siteconfig/clusterinstance_test.go

```diff
@@ -0,0 +1,299 @@
+			namespace:     "",
+			client:        true,
+			exists:        true,
+			expectedError: fmt.Errorf("ClusterInstance 'nsname' cannot be empty"),
+		},
+		{
+			name:          testClusterInstance,
+			namespace:     testClusterInstance,
+			client:        false,
+			exists:        true,
+			expectedError: fmt.Errorf("apiClient cannot be nil"),
+		},
+		{
+			name:      testClusterInstance,
+			namespace: testClusterInstance,
+			client:    true,
+			exists:    false,
+			expectedError: fmt.Errorf(
+				"ClusterInstance object %s does not exist in namespace %s",
+				testClusterInstance, testClusterInstance),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testClient     *clients.Settings
+		)
+
+		if testCase.exists {
+			runtimeObjects = append(runtimeObjects, generateClusterInstance())
+		}
+
+		if testCase.client {
+			testClient = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects:  runtimeObjects,
+				SchemeAttachers: testSchemes,
+			})
+		}
+
+		testBuilder, err := PullClusterInstance(testClient, testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testCase.name, testBuilder.Definition.Name)
+			assert.Equal(t, testCase.namespace, testBuilder.Definition.Namespace)
+		}
+	}
+}
+
+func TestClusterInstanceGet(t *testing.T) {
+	testCases := []struct {
+		exists bool
+	}{
+		{
+			exists: true,
+		},
+		{
+			exists: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+		)
+
+		if testCase.exists {
+			runtimeObjects = append(runtimeObjects, generateClusterInstance())
+		}
+
+		testBuilder := generateClusterInstanceBuilderWithFakeObjects(runtimeObjects)
+
+		clusterinstance, err := testBuilder.Get()
+		if testCase.exists {
+			assert.Nil(t, err)
+			assert.NotNil(t, clusterinstance)
+		} else {
+			assert.NotNil(t, err)
+			assert.Nil(t, clusterinstance)
+		}
+	}
+}
+
+func TestClusterInstanceCreate(t *testing.T) {
+	testCases := []struct {
+		exists bool
+	}{
+		{
+			exists: true,
+		},
+		{
+			exists: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+		)
+
+		if testCase.exists {
+			runtimeObjects = append(runtimeObjects, generateClusterInstance())
+		}
+
+		testBuilder := generateClusterInstanceBuilderWithFakeObjects(runtimeObjects)
+
+		result, err := testBuilder.Create()
+		assert.Nil(t, err)
+		assert.NotNil(t, result)
+		assert.Equal(t, testClusterInstance, result.Definition.Name)
+		assert.Equal(t, testClusterInstance, result.Definition.Namespace)
+	}
+}
+
+func TestClusterInstanceDelete(t *testing.T) {
+	testCases := []struct {
+		exists        bool
+		expectedError error
+	}{
+		{
+			exists:        true,
+			expectedError: nil,
+		},
+		{
+			exists:        false,
+			expectedError: fmt.Errorf("ClusterInstance cannot be deleted because it does not exist"),
```

Comment:
this needs to be updated since delete won't return an error now

Reply:
Good catch. Updated. Thanks!